python/ray/worker.py

```diff
@@ -790,7 +792,6 @@ def _store_outputs_in_object_store(self, object_ids, outputs):
             if isinstance(outputs[i], ray.actor.ActorHandle):
                 raise Exception("Returning an actor handle from a remote "
                                 "function is not allowed).")
-
```

Comment:
What about having an

```python
if outputs[i] is ray.experimental.NoReturn:
    if not self.plasma_client.contains(object_ids[i]):
        raise Exception("Attempting to return 'ray.experimental.NoReturn' from a remote function, but the corresponding ObjectID does not exist in the local object store.")
```

not actually `Exception`, something more specific should be used.

The only issue I can think of is if the task is a long-running task and the object ends up getting evicted before the task returns..

Reply:
I ended up using a `ValueError` for this. Do you think that's specific enough?

Reply:
Changed to RuntimeError because ValueError seems to be intended for the case where an argument to a function has the right type but the wrong value.